app/src/main/kotlin/com/intive/picover/profile/view/ProfileUpdateBottomSheet.kt

```diff
@@ -16,25 +16,34 @@ import androidx.compose.material3.ModalBottomSheet
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextField
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalSoftwareKeyboardController
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import com.intive.picover.R
+import com.intive.picover.common.validator.ValidationStatus
 
 @OptIn(ExperimentalMaterial3Api::class, ExperimentalComposeUiApi::class)
 @Composable
 fun ProfileUpdateBottomSheet(
 	userName: String,
-	updateUserName: (String) -> Unit,
 	updateName: (String) -> Unit,
 	updateVisibility: (Boolean) -> Unit,
-	validateName: (String) -> Boolean,
-	getInvalidNameErrorMessageId: (String) -> Int,
+	onUserNameChange: (String) -> Int?,
 ) {
 	val keyboardController = LocalSoftwareKeyboardController.current
+	var userNameText by remember { mutableStateOf(userName) }
+	var errorMessageId by remember { mutableStateOf(ValidationStatus.ValidText.errorMessageId) }
+	LaunchedEffect(userNameText) {
+		errorMessageId = onUserNameChange(userNameText)
+	}
```

Comment:
I have a feeling that it's a little bit messed up. Current value of the name input should be held in view model and each `onValueChange` from `TextField` should update the value in view model. Each change on the name should also trigger a validation and give us an error as a result. In that way we would get rid of used `LaunchedEffect`.

Reply:
true, i am also a fan of keeping the editable values in VM, and avoid additional launched effects and remembers :D

Reply:
Done